python_testspace_xml/testspace_xml.py

```diff
@@ -145,34 +95,33 @@ def __init__(self, name, status='passed'):
         self.annotations = []
         self.start_time = ""
         self.duration = 0
-        # path to file with additional diagnostics
-        self.diagnostic_file = None
-        self.meta_info = ""
-
-    def _clean_up(self):
-        for a in self.annotations:
-            a._clean_up()
 
     def set_description(self, description):
         self.description = description
 
     def set_duration_ms(self, duration):
         self.duration = duration
 
-    def fail(self, reason):
+    def set_status(self, status):
+        self.status = status
+
+    def set_start_time(self, gmt_string):
+        self.start_time = gmt_string
+
+    def add_fail_annotation(self, message):
```

Comment:
I think the old `fail` method is better as its primary purpose is to fail the case and optionally annotate